### PR TITLE
[Debugger] Let Cocoa Do The Right Thing(tm)

### PR DESCRIPTION
**Pull request by jstedfast**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1000666/
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/999611/

### CODE REVIEW

**Comment by jstedfast** [comment]
@monojenkins backport release-8.4

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs [hunk 1]
```diff
@@ -141,16 +145,36 @@ protected static CGColor GetCGColor (Color color)
...
+			if (baselineOffsets.TryGetValue (key, out baselineOffset))
+				return baselineOffset;
+
+			using (var ctfont = new CTFont (font.FontName, font.PointSize)) {
+				var bbox = ctfont.BoundingBox;
+
+				baselineOffset = (float) (-bbox.Y - ctfont.DescentMetric);
+				baselineOffsets.Add (key, baselineOffset);
+
+				Console.WriteLine ("{0}: baselineOffset={1}", key, baselineOffset);
```
Comment (mrward) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: Should we move this debug code?

Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: yes, thanks

Comment (Therzok) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: Or Debug.WriteLine

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs [hunk 2]
```diff
@@ -141,16 +145,34 @@ protected static CGColor GetCGColor (Color color)
 			return new CGColor ((nfloat) color.Red, (nfloat) color.Green, (nfloat) color.Blue);
 		}
 
-		protected static NSAttributedString GetAttributedString (string text, bool center = false)
+		float GetBaselineOffset (NSFont font)
+		{
+			var key = string.Format (CultureInfo.InvariantCulture, "{0} {1}", font.FontName, font.PointSize);
```
[the remark this answers is not in the document]
Comment (jstedfast) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: or better yet `Dictionary<(string, float), float>`

File: main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs [hunk 3]
```diff
@@ -141,16 +145,34 @@ protected static CGColor GetCGColor (Color color)
 			return new CGColor ((nfloat) color.Red, (nfloat) color.Green, (nfloat) color.Blue);
 		}
 
-		protected static NSAttributedString GetAttributedString (string text, bool center = false)
+		float GetBaselineOffset (NSFont font)
+		{
+			var key = (font.FontName, (float) font.PointSize);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: Do we need to return a float? Maybe just use nfloat all the way?

Comment (Therzok) [reply main/src/addins/MonoDevelop.Debugger/MonoDevelop.Debugger/ObjectValue/Mac/MacDebuggerObjectCellViewBase.cs]: nfloat is IEquatable, so it should be ok

**Comment by sgmunn** [comment]
this is not quite right. When you edit the baselines are off a little and the text drops down a few pixels.

Screen Recording 2019-10-18 at 09 35 AM

**Comment by jstedfast** [comment]
Screen Shot 2019-10-18 at 3 44 52 PM
Screen Shot 2019-10-18 at 3 45 07 PM
Screen Shot 2019-10-18 at 3 58 46 PM
Screen Shot 2019-10-18 at 3 59 10 PM

**Pull request merged**